### PR TITLE
Feat 891 visualization

### LINKED ISSUE #891: visualize Session and Procedures timelines
**Is your feature request related to a problem? Please describe.**
It is difficult to know if a Session or set of Procedures is defined correctly. 

**Describe the solution you'd like**
Add a function that can visualize the timeline of a Session of set of Procedures, including:
- overall start and end time
- start and end times of streams/procedures
- start and end times of stimulus epochs

### PR DESCRIPTION
**Pull request by saskiad**
closes #891

### CODE REVIEW

File: src/aind_data_schema/visualizations/plot_timeline.py [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+def plot_date_of_birth(ax: plt.Axes, date_of_birth: datetime) -> None:
+    """add date of birth marker to the timeline plot"""
     ax.scatter(date_of_birth, [1], marker="o", color="blue", s=100)
     ax.text(date_of_birth, 1.1, "Birth", rotation=90, ha="center", va="bottom")
     ax.text(date_of_birth, 0.9, "Age in days", va="top", ha="right")
 
 
-def plot_procedures(procedures, date_of_birth, ax):
+def plot_procedures(ax: plt.Axes, procedures: dict, date_of_birth: datetime) -> None:
+    """add subject and procedure start and end dates to the timeline plot"""
```
Comment (saskiad) [inline src/aind_data_schema/visualizations/plot_timeline.py:37]: "add procedures start and end dates to the timeline plot."

(remove the "subject and" as the date of birth is in the other function above)

File: src/aind_data_schema/visualizations/plot_timeline.py [hunk 2]
```diff
@@ -29,64 +50,49 @@ def plot_procedures(procedures, date_of_birth, ax):
         ax.text(start_date, 1.1, proc["procedure_name"], rotation=90, ha="center", va="bottom")
 
 
-def plot_date_of_acquisition(date_of_acquisition, date_of_birth, ax):
+def plot_date_of_acquisition(ax: plt.Axes, date_of_acquisition: datetime, date_of_birth: datetime) -> None:
+    """add data of acquisition markers to the timeline plot"""
```
Comment (saskiad) [inline src/aind_data_schema/visualizations/plot_timeline.py:54]: date of acquisition marker

File: src/aind_data_schema/visualizations/plot_timeline.py [hunk 3]
```diff
@@ -29,64 +50,49 @@ def plot_procedures(procedures, date_of_birth, ax):
...
-        ax.text(start_date, 1.1, proc["name"], rotation=90, ha="center", va="bottom")
-
-
-def plot_timeline(datapath, savepath, processing_flag=False):
+def plot_timeline(
+    subject: dict = None,
+    procedures: dict = None,
+    acquisition: dict = None,
+    session: dict = None,
+    processing: dict = None,
```
Comment (saskiad) [inline src/aind_data_schema/visualizations/plot_timeline.py:67]: remove processing

File: src/aind_data_schema/visualizations/plot_timeline.py [hunk 4]
```diff
@@ -29,64 +50,49 @@ def plot_procedures(procedures, date_of_birth, ax):
...
 
     Args:
-        datapath (str): path to folder containing all metadata files
-        savepath (str): path to location to save figure
-        processing_flag (Bool): Making this True adds the processing dates to the timeline. Defaults to False.
+        subject (dict): dictionary containing subject metadata
+        procedures (dict): dictionary containing procedures metadata
+        acquisition (dict): dictionary containing acquisition metadata
+        session (dict): dictionary containing session metadata
+        processing (dict): dictionary containing processing metadata
```
Comment (saskiad) [inline src/aind_data_schema/visualizations/plot_timeline.py:77]: remove processing

**Review by saskiad — COMMENTED** [review]
Looks good!

**Review by dyf — APPROVED** [review]

**Pull request merged**